### PR TITLE
Fix wording, punctuation and Markdown for ServiceAccount concept page

### LINKED ISSUE #28285: No findable description of bound service account tokens
No navigation panel says anything about bound service account tokens.

https://kubernetes.io/docs/reference/access-authn-authz/authentication/ describes service accounts and their non-bound tokens.

https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/ mentions them in a 4-th level subsection under automation in the admission controller (which omits the fact that the kubelet is also involved in automating bound tokens, see #28286 ), but that subsection is omitted from the TOC because it is 4th level.  Apparently that 3rd-level branch is only taking responsibility for describing service account automation in the admission controller.

Bound service account tokens are an aspect of the authentication functionality that should be described on their own right, in the page on authentication.

### PR DESCRIPTION
**Pull request by sftim**
Fixups for PR #38289

Fixes #28285 

Page changed: https://kubernetes.io/docs/concepts/security/service-accounts/

### CODE REVIEW

**Comment by sftim** [comment]
This includes the same fix that's in https://github.com/kubernetes/website/pull/39435

File: content/en/docs/concepts/security/service-accounts.md [hunk 1]
```diff
@@ -189,17 +194,21 @@ to authenticate to the Kubernetes API server, and to any other system where a
 trust relationship exists. Depending on how the token was issued
 (either time-limited using a `TokenRequest` or using a legacy mechanism with
 a Secret), a ServiceAccount token might also have an expiry time, an audience,
-and a time after which the token *starts* being valid. When a process running
-in a Pod attempts to communicate with the Kubernetes API server, it adds an
-`Authorization: Bearer <token>` header to the HTTP request. The API server
-checks the validity of the bearer token as follows:
+and a time after which the token *starts* being valid. When a client that is
+acting as a ServiceAccount tried to communicate with the Kubernetes API server,
```
Comment (shannonxtreme) [inline content/en/docs/concepts/security/service-accounts.md:198]:
```suggestion
acting as a ServiceAccount tries to communicate with the Kubernetes API server,
```

File: content/en/docs/concepts/security/service-accounts.md [hunk 2]
```diff
@@ -189,17 +194,21 @@ to authenticate to the Kubernetes API server, and to any other system where a
...
 1. Check the token signature.
 1. Check whether the token has expired.
-1. Checks whether object references in the token claims are currently valid
+1. Check whether object references in the token claims are currently valid.
 1. Check whether the token is currently valid.
 1. Check the audience claims.
 
+The TokenRequest API produces _bound tokens_ for a ServiceAccount, and that
+binding is linked to the lifetime of a related object such as a Pod that is
+acting as that ServiceAccount.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/security/service-accounts.md:210]:
```suggestion
The TokenRequest API produces _bound tokens_ for a ServiceAccount. This
binding is linked to the lifetime of the client, such as a Pod, that is acting
as that ServiceAccount
```

**Review by shannonxtreme — COMMENTED** [review]
Small suggestions, thank you! 

/lgtm

**Comment by Shubham82** [comment]
Thanks, @sftim for this.
/lgtm

File: content/en/docs/concepts/security/service-accounts.md [hunk 3]
```diff
@@ -189,17 +194,21 @@ to authenticate to the Kubernetes API server, and to any other system where a
...
+request. The API server checks the validity of that bearer token as follows:
 
 1. Check the token signature.
 1. Check whether the token has expired.
-1. Checks whether object references in the token claims are currently valid
+1. Check whether object references in the token claims are currently valid.
 1. Check whether the token is currently valid.
 1. Check the audience claims.
 
+The TokenRequest API produces _bound tokens_ for a ServiceAccount. This
```
Comment (tengqm) [inline content/en/docs/concepts/security/service-accounts.md:208]: bound or bounded?

File: content/en/docs/concepts/security/service-accounts.md [hunk 4]
```diff
@@ -189,17 +194,21 @@ to authenticate to the Kubernetes API server, and to any other system where a
...
+binding is linked to the lifetime of the client, such as a Pod, that is acting
+as that ServiceAccount
```
Comment (tengqm) [inline content/en/docs/concepts/security/service-accounts.md:210]:
```suggestion
as that ServiceAccount.
```

File: content/en/docs/concepts/security/service-accounts.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/concepts/security/service-accounts.md:208]: Bound is right - the tokens have been bound to an object. Bounded would mean that they had a defined scope, whereas we're talking about the existence of an object-token binding. Validity of the token is conditional on the continued existence of the object that the token is bound to.

Comment (tengqm) [reply content/en/docs/concepts/security/service-accounts.md:208]: OK. I was thinking of the expiry time of the token which is bounded.

**Comment by reylejano** [comment]
Re-applying lgtm from @shannonxtreme in [comment](https://github.com/kubernetes/website/pull/39436#issuecomment-1428955524)
/lgtm

**Pull request merged**